modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseRegistry.java

```diff
@@ -298,10 +299,11 @@ void retrieveAndUpdateDatabase(String databaseName, GeoIpTaskState.Metadata meta
             });
     }
 
-    void updateDatabase(String databaseFileName, String recordedMd5, Path file) {
+    void updateDatabase(String databaseFileName, String recordedMd5, Path file, long lastUpdate) {
         try {
             LOGGER.info("database file changed [{}], reload database...", file);
             DatabaseReaderLazyLoader loader = new DatabaseReaderLazyLoader(cache, file, recordedMd5);
+            loader.setLastUpdate(lastUpdate);
```

Comment:
Maybe adjust `DatabaseRegistryTests#testCheckDatabases(...)` to check that last update is set?

Reply:
I've added relevant assertions